web/src/lib/urls.ts

```diff
@@ -18,30 +18,53 @@ export const urls = {
   home: (): "/" => "/",
 
   // ANCHOR: Apps
-  app: (app_id?: string, params?: AppParams): string => {
-    const baseUrl = app_id ? `/app/${app_id}` : "/app";
-
-    if (!params) {
+  app: (props: {
+    app_id?: string;
+    team_id: string;
+    params?: AppParams;
+  }): string => {
+    const baseUrl = props.app_id
+      ? `${urls.team(props.team_id)}/app/${props.app_id}`
+      : `${urls.team(props.team_id)}/app`;
+
+    if (!props.params) {
       return baseUrl;
     }
 
-    const searchParams = new URLSearchParams(params);
+    const searchParams = new URLSearchParams(props.params);
     return `${baseUrl}?${searchParams.toString()}`;
   },
 
-  appSignIn: (app_id?: string): `/app/${string}/sign-in` | "/app/sign-in" =>
-    app_id ? `/app/${app_id}/sign-in` : "/app/sign-in",
-
-  appActions: (app_id?: string): `/app/${string}/actions` | "/app/actions" =>
-    app_id ? `/app/${app_id}/actions` : "/app/actions",
+  appSignIn: (params: {
+    app_id?: string;
+    team_id: string;
+  }): `/team/${string}/app/${string}/sign-in` | `/team/${string}/app/sign-in` =>
+    params.app_id
+      ? `/team/${params.team_id}/app/${params.app_id}/sign-in`
+      : `/team/${params.team_id}/app/sign-in`,
+
+  appActions: (params: {
+    app_id?: string;
+    team_id: string;
+  }): `/team/${string}/app/${string}/actions` | `/team/${string}/app/actions` =>
+    params.app_id
+      ? `/team/${params.team_id}/app/${params.app_id}/actions`
+      : `/team/${params.team_id}/app/actions`,
 
   // ANCHOR: Others
-  kiosk: (action_id: string): `/kiosk/${string}` => `/kiosk/${action_id}`,
-
-  debugger: (app_id: string): `/app/${string}/debugger` =>
-    `/app/${app_id}/debugger`,
-
-  team: (): "/team" => "/team",
+  kiosk: (params: {
+    action_id: string;
+    team_id: string;
+  }): `/team/${string}/kiosk/${string}` =>
+    `/team/${params.team_id}/kiosk/${params.action_id}`,
+
+  debugger: (params: {
+    app_id: string;
+    team_id: string;
+  }): `/team/${string}/app/${string}/debugger` =>
+    `/team/${params.team_id}/app/${params.app_id}/debugger`,
```

Comment:
We are using this debugger url from layout only. There we passing app_id from the URL or in case we can't find it somehow we pass first app from the team's app list. Not sure what should be the case where we don't have even the list of apps.

Reply:
Yeah makes sense there's just an error on the server that might be annoying in the logs
 Invalid href '/team/team_5ce1a9e98fb2402a2085ec76b728239e/app//debugger' passed to next/router in page: '/team/[team_id]'. Repeated forward-slashes (//) or backslashes \ are not valid in the href.